toolkit/ar/src/main/java/com/arcgismaps/toolkit/ar/WorldScaleSceneView.kt

```diff
@@ -310,3 +270,161 @@ public fun WorldScaleSceneView(
+                initializationStatus.update(
+                    WorldScaleSceneViewStatus.FailedToInitialize(
+                        IllegalStateException(
+                            context.getString(
+                                R.string.permissions_not_granted_message,
+                                permissionsNotGranted
+                            )
+                        )
+                    ),
+                    onInitializationStatusChanged
+                )
+            } else {
+                allPermissionsGranted.value = true
+            }
+        }
+
+    LaunchedEffect(Unit) {
+        launcher.launch(permissionsToRequest.toTypedArray())
+    }
+
+    return allPermissionsGranted
+}
+
+/**
+ * Starts the [locationDataSource] and periodically updates the [cameraController] with new locations
+ *
+ * @since 200.7.0
+ */
+@Composable
+private fun LocationTracker(
+    locationDataSource: LocationDataSource,
+    cameraController: TransformationMatrixCameraController,
+    onLocationDataSourceStatus: ((LocationDataSourceStatus)) -> Unit
+) {
+    // We should reset the origin camera if the LDS or camera controller changes
+    var hasSetOriginCamera = remember(locationDataSource, cameraController) { false }
+    LaunchedEffect(locationDataSource) {
+        launch {
+            locationDataSource.status.collect {
+                onLocationDataSourceStatus(it)
+            }
+        }
+        launch {
+            locationDataSource.locationChanged
+                .filter { location ->
+                    !hasSetOriginCamera || shouldUpdateCamera(
+                        location,
+                        cameraController.originCamera.value
+                    )
+                }
+                .collect { location ->
+                    cameraController.setOriginCamera(
+                        Camera(
+                            location.position.y,
+                            location.position.x,
+                            if (location.position.hasZ) location.position.z!! else 0.0,
+                            0.0,
+                            90.0,
+                            0.0
+                        )
+                    )
+                    // We have to do this or the error gets bigger and bigger.
+                    cameraController.transformationMatrix =
+                        TransformationMatrix.createIdentityMatrix()
+                    if (!hasSetOriginCamera) {
+                        hasSetOriginCamera = true
+                    }
+                }
+        }
+    }
+    val lifecycleOwner = LocalLifecycleOwner.current
+    DisposableEffect(locationDataSource) {
+        // This will start the LocationDataSource
+        val wrapper = LocationDataSourceWrapper(locationDataSource)
+        lifecycleOwner.lifecycle.addObserver(wrapper)
+        onDispose {
+            wrapper.onDestroy(lifecycleOwner)
+            lifecycleOwner.lifecycle.removeObserver(wrapper)
```

Comment:
I think it' safer to do it in the opposite order:
```suggestion
            lifecycleOwner.lifecycle.removeObserver(wrapper)
            wrapper.onDestroy(lifecycleOwner)
```